sympy/integrals/risch.py

```diff
@@ -870,7 +870,7 @@ def as_poly_1t(p, t, z):
         # issue 4950
         raise NotImplementedError(e)
     # Compute the negative degree parts.
-    one_t_part = Poly.from_list(reversed(one_t_part.rep.rep), *one_t_part.gens,
+    one_t_part = Poly.from_list(reversed(one_t_part.rep.to_list()), *one_t_part.gens,
```

Comment:
This operation would fail for multivariate polynomials:
```python
In [4]: p = Poly(x**2*y + y, x, y)

In [5]: Poly.from_list(reversed(p.rep.to_list()), *p.gens, domain=p.domain)
---------------------------------------------------------------------------
MultivariatePolynomialError
```
I don't see any particularly better implementation of a method to reverse the coefficients likle this. The only possibility if this was to be better with sparse polynomials would be to use a dict rather than a list but I think it is unlikely to give any speed improvement right now. With `DUP_Python` this just copies the list. With `DUP_Flint` there is currently no way to implement this without calling python-flint's equivalent of `to_list` anyway.

Reply:
See https://github.com/sympy/sympy/issues/5131.